integ-test/src/test/java/org/opensearch/sql/legacy/JoinIT.java

```diff
@@ -228,14 +228,12 @@ public void hintLimits_firstLimitSecondLimitNL() throws IOException {
     hintLimits_firstLimitSecondLimit(true);
   }
 
-  @Ignore("Join limit hint is deprecated and easily broken due to limit on unsorted records")
```

Comment:
Do you know if this test will still be "broken" in some case as the comment said?

Reply:
Same as above, I've run this locally and on multiple GHA runs and it passes. I can create a GitHub issue if this should be reverted back.

Reply:
Same as above. May be safer to leave it if we didn't dig into code and confirmed.

Reply:
Sounds good. Will revert this. Thanks!